include/audio_module/audio_module.h

```diff
@@ -0,0 +1,500 @@
+ *
+ * @param handle[in/out]  The handle of the module that sent the audio data.
+ * @param audio_data[in]  The audio data to operate on.
+ */
+typedef void (*audio_module_response_cb)(struct audio_module_handle_private *handle,
+					 struct audio_data const *const audio_data);
+
+/**
+ * @brief Private pointer to a module's functions.
+ */
+struct audio_module_functions {
+	/**
+	 * @brief Open an audio module with the specified initial configuration.
+	 *
+	 * @note This is an optional function for an audio module.
+	 *
+	 * @note This is a mandatory function for all audio modules.
+	 * @param handle[in/out]     The handle to the module instance.
+	 * @param configuration[in]  Pointer to the desired initial configuration to set.
+	 *
+	 * @return 0 if successful, error otherwise.
+	 */
+	int (*open)(struct audio_module_handle_private *handle,
+		    struct audio_module_configuration const *const configuration);
+
+	/**
+	 * @brief Close an open audio module.
+	 *
+	 * @note This is an optional function for an audio module.
+	 *
+	 * @param handle[in/out]  The handle to the module instance.
+	 *
+	 * @return 0 if successful, error otherwise.
+	 */
+	int (*close)(struct audio_module_handle_private *handle);
+
+	/**
+	 * @brief Reconfigure an audio module after it has been opened with its initial
+	 *        configuration.
+	 *
+	 * @note This is a mandatory function for an audio module.
+	 *
+	 * @param handle[in/out]     The handle to the module instance.
+	 * @param configuration[in]  Pointer to the desired configuration to set.
+	 *
+	 * @return 0 if successful, error otherwise.
+	 */
+	int (*configuration_set)(struct audio_module_handle_private *handle,
+				 struct audio_module_configuration const *const configuration);
+
+	/**
+	 * @brief Get the configuration of an audio module.
+	 *
+	 * @note This is a mandatory function for an audio module.
+	 *
+	 * @param handle[in]          The handle to the module instance.
+	 * @param configuration[out]  Pointer to the module's current configuration.
+	 *
+	 * @return 0 if successful, error otherwise.
+	 */
+	int (*configuration_get)(struct audio_module_handle_private const *const handle,
+				 struct audio_module_configuration *configuration);
+
+	/**
+	 * @brief Start an audio module.
+	 *
+	 * @note This is an optional function for an audio module.
+	 *
+	 * @note This is an optional function for an audio module.
+	 * @param handle[in/out]  The handle for the module to start.
+	 *
+	 * @return 0 if successful, error otherwise.
+	 */
+	int (*start)(struct audio_module_handle_private *handle);
+
+	/**
+	 * @brief Stop an audio module.
+	 *
+	 * @note This is an optional function for an audio module.
+	 *
+	 * @param handle[in/out]  The handle for the module to stop.
+	 *
+	 * @return 0 if successful, error otherwise.
+	 */
+	int (*stop)(struct audio_module_handle_private *handle);
+
+	/**
+	 * @brief The core data processing for an audio module. Can be either an
```

Comment:
I suggest formatting it like this:
![image](https://github.com/gWacey/sdk-nrf/assets/89020256/f77990be-d273-40bf-8a9b-e95ad1bdc112)